src/ophyd_async/fastcs/panda/_table.py

```diff
@@ -34,137 +36,88 @@ class SeqTrigger(str, Enum):
-        time2=time2,
-        outa2=or_default(outa2, np.bool_),
-        outb2=or_default(outb2, np.bool_),
-        outc2=or_default(outc2, np.bool_),
-        outd2=or_default(outd2, np.bool_),
-        oute2=or_default(oute2, np.bool_),
-        outf2=or_default(outf2, np.bool_),
-    )
-    for k, v in table.items():
-        size = len(v)  # type: ignore
-        if size != length:
-            raise ValueError(f"{k}: has length {size} not {length}")
-    return table
+PydanticNp1DArrayInt32 = Annotated[
+    np.ndarray[tuple[int], np.int32],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.int32, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], np.int32)),
+]
+PydanticNp1DArrayBool = Annotated[
+    np.ndarray[tuple[int], np.bool_],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.bool_, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], dtype=np.bool_)),
+]
+
+PydanticNp1DArrayUnicodeString = Annotated[
+    np.ndarray[tuple[int], np.unicode_],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.unicode_, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], dtype=np.dtype("<U32"))),
+]
+
+
+class SeqTable(PvaTable):
+    repeats: PydanticNp1DArrayInt32
+    trigger: PydanticNp1DArrayUnicodeString
+    position: PydanticNp1DArrayInt32
+    time1: PydanticNp1DArrayInt32
+    outa1: PydanticNp1DArrayBool
+    outb1: PydanticNp1DArrayBool
+    outc1: PydanticNp1DArrayBool
+    outd1: PydanticNp1DArrayBool
+    oute1: PydanticNp1DArrayBool
+    outf1: PydanticNp1DArrayBool
+    time2: PydanticNp1DArrayInt32
+    outa2: PydanticNp1DArrayBool
+    outb2: PydanticNp1DArrayBool
+    outc2: PydanticNp1DArrayBool
+    outd2: PydanticNp1DArrayBool
+    oute2: PydanticNp1DArrayBool
+    outf2: PydanticNp1DArrayBool
+
+    @classmethod
+    def row(
+        cls,
+        *,
+        repeats: int = 0,
+        trigger: str = "",
+        position: int = 0,
+        time1: int = 0,
+        outa1: bool = False,
+        outb1: bool = False,
+        outc1: bool = False,
+        outd1: bool = False,
+        oute1: bool = False,
+        outf1: bool = False,
+        time2: int = 0,
+        outa2: bool = False,
+        outb2: bool = False,
+        outc2: bool = False,
+        outd2: bool = False,
+        oute2: bool = False,
+        outf2: bool = False,
+    ) -> "SeqTable":
+        return PvaTable.row(
+            cls,
+            repeats=repeats,
+            trigger=trigger,
+            position=position,
+            time1=time1,
+            outa1=outa1,
+            outb1=outb1,
+            outc1=outc1,
+            outd1=outd1,
+            oute1=oute1,
+            outf1=outf1,
+            time2=time2,
+            outa2=outa2,
+            outb2=outb2,
+            outc2=outc2,
+            outd2=outd2,
+            oute2=oute2,
+            outf2=outf2,
+        )
```

Comment:
As we are dropping typing here, might as well make the code shorter. If we made `PvaTable.row` a `@staticmethod`, then would this work?
```suggestion
        return PvaTable.row(**locals())
```

Reply:
`locals()` here includes a lot more than the kwargs we want. I changed it to the following:
``` python
        sig = inspect.signature(cls.row)
        kwargs = {k: v for k, v in locals().items() if k in sig.parameters}
        return Table.row(cls, **kwargs)
```

Reply:
> `locals()` here includes a lot more than the kwargs we want. 

Do you get more than this?

```python
>>> class A:
...     @classmethod
...     def meth(cls, x=1, b=2):
...             print(locals())
... 
>>> A.meth()
{'cls': <class '__main__.A'>, 'x': 1, 'b': 2}
```